apps/test_ogrsf.cpp

```diff
@@ -401,6 +401,40 @@ static int TestDataset( GDALDriver** ppoDriver )
     poDS->GetMetadata( nullptr );
     poDS->GetMetadataItem( "", nullptr );
 
+    if ( !poDriver->GetMetadataItem( GDAL_DCAP_VECTOR ))
+    {
+        printf("FAILURE: Driver does not advertize GDAL_DCAP_VECTOR!\n" );
+        GDALClose(poDS);
```

Comment:
instead of returning, just set bRet = false so that later checks can be also done in the same run. Same for below checks